jwst/master_background/master_background_step.py

```diff
@@ -231,14 +231,21 @@ def split_container(container):
     """Divide a ModelContainer with science and background into one of each
     """
     asn = container.meta.asn_table.instance
+
     background = datamodels.ModelContainer()
     science = datamodels.ModelContainer()
-    for product in asn['products']:
-        for member in product['members']:
-            if member['exptype'].lower() == 'science':
-                science.append(datamodels.open(member['expname']))
-            if member['exptype'].lower() == 'background':
-                background.append(datamodels.open(member['expname']))
+
+    ind_science = container.ind_asn_type('science')
```

Comment:
Why are these numpy arrays as opposed to just lists? This does not seem necessary? Is there some type of math happening on `DataModel`s?  This would allow a more concise code something like:
```
for ind_science in container.ind_asn_type('science'):
    science.append(container._models[ind_science])
```

Reply:
They were coded as numpy arrays prior to my meddling, and I wasn't sure why - so in fear of breaking something, I left it. I'll just tear off the bandaid and see if any pain surfaces.